cmd/kogito/deploy_data_index.go

```diff
@@ -0,0 +1,150 @@
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/spf13/cobra"
+)
+
+type deployDataIndexFlags struct {
+	deployCommonFlags
+	image              string
+	kafka              v1alpha1.KafkaConnectionProperties
+	infinispan         v1alpha1.InfinispanConnectionProperties
+	infinispanSasl     string
+	infinispanUser     string
+	infinispanPassword string
+}
+
+const (
+	defaultDataIndexName         = "kogito-data-index"
+	defaultInfinispanSecretName  = "kogito-data-index-infinispan-credentials"
+	defaulInfinispanUsernameKey  = "username"
+	defaultInfinispanPasswordKey = "password"
+)
+
+var (
+	deployDataIndexCmd      *cobra.Command
+	deployDataIndexCmdFlags = deployDataIndexFlags{
+		deployCommonFlags: deployCommonFlags{},
+		kafka:             v1alpha1.KafkaConnectionProperties{},
+		infinispan:        v1alpha1.InfinispanConnectionProperties{},
+	}
+)
+
+var _ = RegisterCommandVar(func() {
+	deployDataIndexCmd = &cobra.Command{
+		Use:   "deploy-data-index [flags]",
+		Short: "Deploys the Kogito Data Index Service in the given Project",
+		Long: `deploy-data-index will deploy the Data Index service to enable capturing and indexing data produced by one or more Kogito Runtime Services.
+				For more information see: https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deployDataIndexExec(cmd, args)
+		},
+		PreRun:  preRunF,
+		PostRun: posRunF,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(deployDataIndexCmdFlags.infinispanUser) > 0 && len(deployDataIndexCmdFlags.infinispanPassword) == 0 {
+				return fmt.Errorf("Please set infinispan-user and infinispan-password")
+			}
+			if len(deployDataIndexCmdFlags.infinispanUser) == 0 && len(deployDataIndexCmdFlags.infinispanPassword) > 0 {
+				return fmt.Errorf("Please set infinispan-user and infinispan-password")
+			}
+			if len(deployDataIndexCmdFlags.infinispanUser) > 0 &&
+				len(deployDataIndexCmdFlags.infinispanPassword) > 0 &&
+				len(deployDataIndexCmdFlags.infinispanSasl) == 0 {
+				deployDataIndexCmdFlags.infinispanSasl = string(v1alpha1.SASLPlain)
+			}
+			if err := commonCheckDeployArgs(&deployDataIndexCmdFlags.deployCommonFlags); err != nil {
+				return err
+			}
+			if err := commonCheckImageTag(deployDataIndexCmdFlags.image); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+})
+
+var _ = RegisterCommandInit(func() {
+	rootCmd.AddCommand(deployDataIndexCmd)
+	commonAddDeployFlags(deployDataIndexCmd, &deployDataIndexCmdFlags.deployCommonFlags)
+
+	deployDataIndexCmd.Flags().StringVarP(&deployDataIndexCmdFlags.image, "image", "i", resdataindex.DefaultImage, "Image tag (namespace/name:tag) for the runtime Service, e.g: openshift/kogito-data-index:latest")
+	deployDataIndexCmd.Flags().StringVar(&deployDataIndexCmdFlags.kafka.ServiceURI, "kafka-url", "", "The Kafka cluster internal URI, example: my-kafka-cluster:9092")
```

Comment:
I think it may be good to mention which properties are mandatory and which are optional, so users don't have to find it out by themselves.